src/libraries/Common/src/System/Net/IPv4AddressHelper.Common.cs

```diff
@@ -105,32 +113,35 @@ internal static unsafe bool IsValid(char* name, int start, ref int end, bool all
         //                 / "2" %x30-34 DIGIT     ; 200-249
         //                 / "25" %x30-35          ; 250-255
         //
-        internal static unsafe bool IsValidCanonical(char* name, int start, ref int end, bool allowIPv6, bool notImplicitFile)
+        internal static bool IsValidCanonical(ReadOnlySpan<TChar> name, ref int bytesConsumed, bool allowIPv6, bool notImplicitFile)
```

Comment:
```suggestion
        internal static bool IsValidCanonical(ReadOnlySpan<TChar> name, out int bytesConsumed, bool allowIPv6, bool notImplicitFile)
```

Reply:
Accepting both a `ReadOnlySpan` argument and a `ref int bytesConsumed` is odd. It's not obvious to the caller whether the argument should be initialized to anything vs. it'll always be output-only.

It was previously a `ref` because it also acted as the input argument for the length.